common/types.go

```diff
@@ -234,16 +238,289 @@ func init() {
 }
 
 // Address represents the 20 byte address of an Ethereum account.
-type Address [AddressLength]byte
+type InternalAddress [AddressLength]byte
+type ExternalAddress [AddressLength]byte
+
+type Address struct {
```

Comment:
I'm trying to remember the rational for this. Originally we discussed converting the `Address` type to an interface:
```
type Address interface{
    IsInternal() bool
    IsExternal() bool
    ...
}
```
and everywhere that operates on addresses would have to explicitly perform a type assertion + cast to use the address, e.g:
```
func someInternalAddressLogic(addr Address) {
    a, ok := addr.(InternalAddress) {
    if !ok {
        ... error I need an internal address! ...
    }
    ... my logic ...
}
```

BUT, I remember we had a conversation about using a struct instead of an interface, because there was some associated data that needed to be stored in the struct... but I don't see any associated data in the struct, so I'm confused. Where did we land on all this? I would definitely prefer an interface with explicit type assertions if possible

Reply:
Was this comment resolved?